test/e2e/helpers.go

```diff
@@ -64,3 +72,126 @@ func verifyHealthStatusMetric(caCtx *test.Context, metricsPath string, metricLab
 	}
 }
+
+func setupServingControlPlaneMetricsRoutes(caCtx *test.Context, serviceMonitorServiceName string) (*v1.Route, error) {
+	routeName := serviceMonitorServiceName
+	metricsRoute := &v1.Route{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      routeName,
+			Namespace: servingNamespace,
+		},
+		Spec: v1.RouteSpec{
+			Port: &v1.RoutePort{
+				TargetPort: intstr.FromString("8444"),
+			},
+			To: v1.RouteTargetReference{
+				Kind: "Service",
+				Name: serviceMonitorServiceName,
+			},
+			TLS: &v1.TLSConfig{
+				Termination: v1.TLSTerminationReencrypt,
+			},
+		}}
+	r, err := caCtx.Clients.Route.Routes(servingNamespace).Create(context.Background(), metricsRoute, meta.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error creating service monitor route: %v", err)
+	}
+	caCtx.AddToCleanup(func() error {
+		caCtx.T.Logf("Cleaning up service metrics route: %s", routeName)
+		return caCtx.Clients.Route.Routes("openshift-serverless").Delete(context.Background(), routeName, meta.DeleteOptions{})
+	})
+	return r, nil
+}
+
+func verifyServingControlPlaneMetrics(caCtx *test.Context) {
+	serviceMonitorsInstances := []struct {
+		name        string
+		expectedStr string
+	}{{
+		name:        "activator-sm",
+		expectedStr: `activator_client_results{name=""}`,
+	}, {
+		name:        "autoscaler-sm",
+		expectedStr: "autoscaler_actual_pods",
+	}, {
+		name:        "autoscaler-hpa-sm",
+		expectedStr: "hpaautoscaler_client_latency_bucket",
+	}, {
+		name:        "controller-sm",
+		expectedStr: "controller_client_latency_bucket",
+	}, {
+		name:        "domain-mapping-sm",
+		expectedStr: "domainmapping_client_latency_bucket",
+	}, {
+		name:        "domainmapping-webhook-sm",
+		expectedStr: "domainmapping_webhook_client_latency_bucket",
+	}, {
+		name:        "webhook-sm",
+		expectedStr: "webhook_client_latency_bucket",
+	}}
+	for _, sm := range serviceMonitorsInstances {
+		serviceName := sm.name + "-service"
+		_, err := caCtx.Clients.Kube.CoreV1().Services(servingNamespace).Get(context.Background(), serviceName, meta.GetOptions{})
+		if err != nil {
+			caCtx.T.Fatalf("Error getting service monitor service: %v", err)
+		}
+		route, err := setupServingControlPlaneMetricsRoutes(caCtx, serviceName)
+		if err != nil {
+			caCtx.T.Fatalf("Failed to setup operator metrics route: %v", err)
+		}
+		metricsPath := "https://" + route.Spec.Host + "/metrics"
+		metricsURL, err := url.Parse(metricsPath)
+		if err != nil {
+			caCtx.T.Fatalf("Error parsing url for metrics: %v", err)
+		}
+		bToken := getBearerTokenForAuthorizedAccount(caCtx)
+		var reqOption pkgTest.RequestOption = func(request *http.Request) {
+			request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", bToken))
+		}
+		var transportOption spoof.TransportOption = func(transport *http.Transport) *http.Transport {
+			transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
Using the `reencrypt` option here plus avoiding validating the crt at the route. The connection from the route to the service is validated from the OCP check this [great source](https://www.openshift.com/blog/self-serviced-end-to-end-encryption-approaches-for-applications-deployed-in-openshift) for more info. Also in the case of a service using a service signing certificate (as in our case) there is no need to have a destination CA for the route.
Passthrough would not work here as we dont have a certifacate with two-SANS (we only generate certs for cluster internal use) and using this option will cause an error:
```
TestKnativeServing: spoof.go:175: NOT Retrying https://activator-sm-service-knative-serving.apps.ci-ln-98rqthb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/metrics: Get "https://activator-sm-service-knative-serving.apps.ci-ln-98rqthb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/metrics": x509: certificate is valid for activator-sm-service.knative-serving.svc, activator-sm-service.knative-serving.svc.cluster.local, not activator-sm-service-knative-serving.apps.ci-ln-98rqthb-d5d6b.origin-ci-int-aws.dev.rhcloud.com```